### ISSUE TITLE
[Bug] The hint for hint-manifest-file-extension should allow .json file extension and should check for the MIME type

### ISSUE THREAD

**Issue by thejohnjansen** [webhintio/hint#2877]
# 🐛 Bug report

## Description
1. browse to https://webhint.io and run a scan on https://outweb.io (or any other PWA)

NOTICE: In the PWA section, it will incorrectly say that the file extension type for the manifest is wrong and must be .webmanifest instead of .json
EXPECT: There is an [issue](https://github.com/w3c/manifest/issues/689#issuecomment-450510898) on the spec to be more clear that .json is fine as long as the MIME type is application/manifest+json

I recommend that the hint allow both .webmanifest and .json as long as the MIME type is application/manifest+json

## Environment
Just using the online scanner

## `webhint` configuration
Default

**Comment by molant** [comment]
From the docs:

> While the .webmanifest file extension is not enforced by the specification, nor is it required by browsers, using it makes it:
> * easier to set custom server configurations for the web app manifest file
> * possible to benefit from existing configurations

https://webhint.io/docs/user-guide/hints/hint-manifest-file-extension/#why-is-this-important

We will have to decide how to reconcialize this.




________________________________

**Comment by molant** [comment]
Having 2 mime types for the same extension is not something that can be achieved easily. With iis you probably have to use a rewrite rule with the possible perf impact.


________________________________

**Comment by antross** [comment]
@molant - What do you think about adding a new `Severity` value of `suggestion` for this?

Currently we have:
```ts
export enum Severity {
    off = 0,
    warning = 1,
    error = 2
}
```

VS Code offers `Error`, `Warning`, `Information`, and `Hint`. I'd probably recommend mapping this one to `Hint` in that context. We can discuss different displays for other contexts such as the browser extension.

**Comment by molant** [comment]
I think this should be part of the broader conversation we want to have around configurations and such. Aligning with VS Code might be a good option for familiarity and to avoid having to do more mappings.

We should lock something in our calendars to talk about all of this. Maybe we should create a new epic for the next version with possible breaking changes?

**Comment by antross** [comment]
Agree that's a good conversation to have, but I think we can consider a new severity level independently. We're going to have to do some mapping for VS Code regardless so I don't think we need to make a breaking change for this specific scenario.

**Comment by molant** [comment]
Maybe not a breaking change but it will be good to have a roadmap of the changes we want to add and what hints should be impacted. One of the pieces of feedback we've heard the most is that people sometimes don't know where to start and what matters the most so I can see this as an opportunity to improve that. Do we just need a `Hint` severity? Maybe adding `Information` is good to. Maybe a hint can report several different severity messages and with the configuration you could decide the minimum level you want to get notified?

There are many options and I think this deserves an RFC of its own tbh to rationalize everything.

**Comment by antross** [comment]
Update: This is being refined in #3288 as part of the severity work. Going forward this rule will report files with `.json` as a `Hint` and other non-`.webmanifest` extensions as a `Warning`.

**Comment by Malvoz** [comment]
Admittedly I haven't read this great wall of text, but consider that **any file extension is a valid one** as long as the `Content-Type` is appropriate (e.g. you could potentially serve `manifest.teapot` with either `application/manifest+json`, `application/json`, or potentially even another JSON-based MIME-type extension (there are plenty: https://www.iana.org/assignments/media-types/media-types.xhtml) but that's probably best avoided.

**Comment by antross** [comment]
> any file extension is a valid one as long as the Content-Type is appropriate

I agree that's technically allowed, but is it a good idea? I think this may be a case where we want to continue to recommend a "best practice" that goes beyond the specification. Especially if you pick a file extension that is typically associated with some other known `Content-Type` it seems problematic. E.g. `.manifest` in particular comes to mind as one to avoid...
